mlflow/xgboost.py

```diff
@@ -187,9 +188,23 @@ def save_model(
 
     conda_env_subpath = "conda.yaml"
     if conda_env is None:
-        pip_requirements = _parse_pip_requirements(pip_requirements)
-        extra_pip_requirements = _parse_pip_requirements(extra_pip_requirements)
-        pip_reqs = pip_requirements or (_get_default_pip_requirements() + extra_pip_requirements)
+        # TODO: Consider pulling these lines out in a function
+        # ====================================================
+        if pip_requirements is not None:
+            pip_reqs, constraints = _parse_pip_requirements(pip_requirements)
+        elif extra_pip_requirements is not None:
+            extra_pip_requirements, constraints = _parse_pip_requirements(extra_pip_requirements)
+            pip_reqs = _get_default_pip_requirements() + extra_pip_requirements
+        else:
+            constraints = None
+            pip_reqs = _get_default_pip_requirements()
+
+        if constraints:
+            pip_reqs.append(f"-c {_CONSTRAINTS_FILE_NAME}")
+            with open(os.path.join(path, _CONSTRAINTS_FILE_NAME), "w") as f:
+                f.write("\n".join(constraints))
+        # ====================================================
+
```

Comment:
I noticed that the current implementation doesn't parse requirements and constraints files in `conda_env`.

```yml
name: my-model
channels:
  - conda-forge
dependencies:
  - python=3.6
  - pip
  - pip:
      - mlflow>=1.6.0
      - xgboost
      - -r requirements.txt
      - -c constraints.txt
```

Reply:
I'll fix this.